notification-extension/ITBNotificationServiceExtension.swift

```diff
@@ -4,15 +4,31 @@
 
 import UserNotifications
 
+public protocol ITBNotificationServiceExtensionDelegate {
+    func notificationServiceDidReceive(_ request: UNNotificationRequest, bestAttemptContent: UNMutableNotificationContent, contentHandler: @escaping (UNNotificationContent) -> Void)
+}
+
 @objc open class ITBNotificationServiceExtension: UNNotificationServiceExtension {
     var contentHandler: ((UNNotificationContent) -> Void)?
     var bestAttemptContent: UNMutableNotificationContent?
+    public static var itblNotificationDelegate: ITBNotificationServiceExtensionDelegate?
     
     @objc override open func didReceive(_ request: UNNotificationRequest,
                                         withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
         self.contentHandler = contentHandler
         bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
         
+        if let itblNotificationDelegate = ITBNotificationServiceExtension.itblNotificationDelegate,
+           let bestAttemptContent = bestAttemptContent {
+            
+            let result = request.content.userInfo.contains { $0.key as! String == JsonKey.Payload.metadata }
```

Comment:
@christinaschell does it make sense to add a failure condition here for casting the key to a `String`? I'm thinking if the cast isn't successful, we print an error to the console and just `return`. I'm thinking that, in the case this cast doesn't succeed, we don't want it to crash here, specifically.